### PR TITLE
storage: disable crashing for single delete callbacks

### PR DESCRIPTION
**Pull request by sumeerbhola**
As elaborated in code comments, these can be false positives caused by delete-only compactions. The metrics are not disabled, under the rare chance that we see a problem in a cluster and one of the metrics (if it happens to not be a false positive) gives us a hint about the cause of the problem. For the same reason, we log such callbacks every 5 minutes, so we can correlate the key in such logs with an actual problem. False positives should be rare, especially for the invariant violation callback.

Informs https://github.com/cockroachdb/cockroach/issues/115881
Informs https://github.com/cockroachdb/cockroach/issues/114421

Epic: none

Release note (ops change): The cluster settings
storage.single_delete.crash_on_invariant_violation.enabled and storage.single_delete.crash_on_ineffectual.enabled are disabled and must not be enabled.

### CODE REVIEW

File: pkg/storage/pebble.go [hunk 1]
```diff
@@ -166,19 +166,75 @@ var IngestAsFlushable = settings.RegisterBoolSetting(
...
 	settings.WithPublic,
 )
 
 var SingleDeleteCrashOnIneffectual = settings.RegisterBoolSetting(
 	settings.SystemOnly,
 	"storage.single_delete.crash_on_ineffectual.enabled",
 	"set to true to crash if the single delete was ineffectual",
-	true,
+	false,
 	settings.WithPublic,
```
Comment (rafiss) [inline pkg/storage/pebble.go:238]: if these must not be enabled, then should we make them non-public? if they remain public, should the cluster setting description say that they must not be enabled?

**Review by sumeerbhola — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116889)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @jbowens and @rafiss)

___
*[`pkg/storage/pebble.go` line 238 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/116889#gh-1433190412:-Nm86W6Q4z4adKVaBgoa:b-niy1pz) ([raw file](https://github.com/cockroachdb/cockroach/blob/56d4ed8c7112cb0ae6d3def8e12aef3ce39f916e/pkg/storage/pebble.go#L238)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

if these must not be enabled, then should we make them non-public? if they remain public, should the cluster setting description say that they must not be enabled?
</blockquote></details>

Changed these to explicitly say Reserved.

**Review by jbowens — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116889#-:-Nm87nco5IOLvaXu-AQp:b593vm9)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss and @sumeerbhola)

___
*[`pkg/storage/pebble.go` line 195 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/116889#-Nm87RMyBQgSn6ZAWaaw:-Nm87RMyBQgSn6ZAWaax:bbgsq69) ([raw file](https://github.com/cockroachdb/cockroach/blob/56d4ed8c7112cb0ae6d3def8e12aef3ce39f916e/pkg/storage/pebble.go#L195)):*
> ```Go
> // SET g#20 in L6
> //
> // In this example, the two SETs represent the same intent, and the RANGEDELa
> ```

nit: `RANGEDELs`

**Review by sumeerbhola — COMMENTED** [review]
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116889#-:-Nm88R243tNz0HHAG2H4:bz5mair)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @jbowens and @rafiss)

___
*[`pkg/storage/pebble.go` line 195 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/116889#-Nm87RMyBQgSn6ZAWaaw:-Nm88PiP7Yrak775RtRh:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/56d4ed8c7112cb0ae6d3def8e12aef3ce39f916e/pkg/storage/pebble.go#L195)):*
<details><summary><i>Previously, jbowens (Jackson Owens) wrote…</i></summary><blockquote>

nit: `RANGEDELs`
</blockquote></details>

Done

**Comment by sumeerbhola** [comment]
test failure is unrelated https://github.com/cockroachdb/cockroach/issues/116837

**Pull request merged**